lib/contentful/query.ex

```diff
@@ -266,11 +271,132 @@ defmodule Contentful.Query do
           queryable
       end
 
-    {url, api_key |> Delivery.request_headers()}
+    {url, api_key |> Request.headers()}
     |> Delivery.send_request()
     |> Delivery.parse_response(&queryable.resolve_entity_response/1)
   end
 
+  @doc """
+  Adds a filter condition to the query.
+
+  This will work for Entries *requiring* a call to `content_type` before:
+
+  ## Example
+
+      import Contentful.Query
+      alias Contentful.Delivery.Entries
+
+      {:ok, entries, total: 1}
+      = Entries
+        |> content_type("dogs")
+        |> by(name: "Hasso", breed: "dalmatian")
+        |> fetch_all
+
+  This will also allow for more complex queries using modifiers:
+
+  ## Example
+
+      import Contentful.Query
+      alias Contentful.Delivery.Entries
+
+      {:ok, entries, total: 100}
+      = Entries
+        |> content_type("dogs")
+        |> by(name: [ne: "Hasso"], breed: "dalmatian")
+        |> fetch_all
+
+  Allowed modifiers are `[:in, :nin, :ne, :lte, :gte, :lt, :gt, :match, :exist]`.
+
+  Working with `Contentful.Delivery.Assets` requires no `content_type` call:
+
+  ## Example
+
+      import Contentful.Query
+      alias Contentful.Delivery.Assets
+
+      {:ok, assets, total: 1} = Assets |> by(id: "foobar") |> fetch_all
+
+  Calling `by/2` allows for adding multiple conditions to the query:
+
+  ## Example
+
+      import Contentful.Query
+      alias Contentful.Delivery.Assets
+
+      {:ok, assets, total: 200}
+      = Assets
+        |> by(tags: [nin: "maps"])
+        |> fetch_all
+
+  """
```

Comment:
Maybe let's add some links to the official documentation, too?

Reply:
I've added it further above